traffic_ops/traffic_ops_golang/cachegroup/cachegroups.go

```diff
@@ -379,6 +379,45 @@ func (cg *TOCacheGroup) deleteCoordinate(coordinateID int) error {
 	return nil
 }
 
+func GetCacheGroupsByName(names []string, Tx *sqlx.Tx) (map[string]tc.CacheGroupNullable, error, error, int) {
+	query := SelectQuery() + multipleCacheGroupsWhere()
+	namesPqArray := pq.Array(names)
+	rows, err := Tx.Query(query, namesPqArray)
+	if err != nil {
+		userErr, sysErr, errCode := api.ParseDBError(err)
+		return nil, userErr, sysErr, errCode
+	}
```

Comment:
below this `if` block we need to `defer rows.Close()` in order to ensure that the rows are properly closed however this function may exit

Reply:
Deferring in 73fb323465.